### PR TITLE
Switch to Webpack

### PR DESCRIPTION
**Pull request by johartl**
Advantages of webpack (compared to browserify)
- Very customizable through `webpack.config.js` and lots of plugins
- Huge optimization potential
- Relative paths for templates and stylesheets
- Using scoped styles rather than importing all scss files into `app.core.scss` (@MajorBreakfast!!)
- Support for build variables (like `BUILD_ENV === 'develop'`)
- Cache buster for bundled files set up

### CODE REVIEW

File: ionic2/app/components/map/map.component.ts [hunk 1]
```diff
@@ -9,8 +9,10 @@ PokeMap.prototype.on = function(...args) {console.debug('map:on', ...args)};
 PokeMap.prototype.goTo = function(...args) {console.debug('map:goTo', ...args)};
 PokeMap.prototype.updateTimeRange = function(...args) {console.debug('map:updateTimeRange', ...args)};
 
-@Directive({
-  selector: 'map'
+@Component({
+  selector: 'map',
+  template: '',
+  styles: [require('./map.component.scss')]
```
Comment (WoH) [inline ionic2/app/components/map/map.component.ts]: shoudl be styleUrl:´require('./map.component.scss')´

File: ionic2/app/components/nav-sidebar/nav-sidebar.component.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ import { ImprintPage } from '../../pages/imprint/imprint.page';
 
 @Component({
   selector: 'nav-sidebar',
-  templateUrl: 'components/nav-sidebar/nav-sidebar.component.html'
+  template: require('./nav-sidebar.component.html'),
+  styles: [require('./nav-sidebar.component.scss')]
```
Comment (WoH) [inline ionic2/app/components/nav-sidebar/nav-sidebar.component.ts]: see above

**Comment by WoH** [comment]
Seems to be working as intended.
I'll try a codova build next and check the code itself.

**Comment by WoH** [comment]
Cordova builds still fail, should be good to go anyway

**Comment by johartl** [comment]
I will merge this now. But out of curiosity, what's the command I need to run to trigger the cordova build?

**Comment by WoH** [comment]
https://github.com/PokemonGoers/Catch-em-all/issues/56

**Pull request merged**